Add note about `--no-server-header` limitations when using WebSockets implementations (#1575)
Kludex/uvicorn: docs/settings.md

* `--lifespan <str>` - Set the Lifespan protocol implementation. **Options:** *'auto', 'on', 'off'.* **Default:** *'auto'*.
* `--h11-max-incomplete-event-size <int>` - Set the maximum number of bytes to buffer of an incomplete event. Only available for `h11` HTTP protocol implementation. **Default:** *'16384'* (16 KB).

> 

Reviewer: ```suggestion
```

* `--lifespan <str>` - Set the Lifespan protocol implementation. **Options:** *'auto', 'on', 'off'.* **Default:** *'auto'*.
* `--h11-max-incomplete-event-size <int>` - Set the maximum number of bytes to buffer of an incomplete event. Only available for `h11` HTTP protocol implementation. **Default:** *'16384'* (16 KB).

!!! note

>     On selecting `websockets` as Websocket protocol, the server headers are not disabled even if `--no-server-header` settings is passed. This is a limitation. 

Reviewer: Instead of being here, it could be where the `--no-server-header` is mentioned, right? :thinking: 
```suggestion
    The `--no-server-header` flag doesn't have effect on the WebSockets implementations.
```

Author: Sure, it makes sense! Accepted your suggestion! 

Thanks!!
